src/store/ProductsStore/ProductsStore.ts

```diff
@@ -0,0 +1,116 @@
+import { API_ENDPOINTS } from "@configs/api";
+import { Product } from "@pages/ProductDetail";
+import rootStore from "@store/RootStore/instance";
+import { Meta } from "@utils/meta";
+import { ILocalStore } from "@utils/useLocalStore";
+import axios from "axios";
+import {
+  action,
+  computed,
+  IReactionDisposer,
+  makeObservable,
+  observable,
+  reaction,
+  runInAction,
+} from "mobx";
+
+type searchTitle = undefined | string | string[] | qs.ParsedQs | qs.ParsedQs[];
+
+type PrivateFields =
+  | "_totalProductsList"
+  | "_meta"
+  | "_searchTitle"
+  | "_productsPage"
+  | "_hasMoreData";
+
+export default class ProductsStore implements ILocalStore {
+  private _totalProductsList: Product[] = [];
+  private _meta: Meta = Meta.initial;
+  private _searchTitle: searchTitle = "";
+  private _productsPage: number = 0;
+  private _hasMoreData: boolean = true;
```

Comment:
некрит, то глаз мылит определение типов там, где это не нужно - ts сам подхватит, да и человек со стороны догадается, что true – это boolean
```suggestion
  private _meta= Meta.initial;
  private _searchTitle: searchTitle = "";
  private _productsPage= 0;
  private _hasMoreData = true;
```